client/src/app/pages/applications/applications-table/applications-table.tsx

```diff
@@ -182,43 +205,36 @@ export const ApplicationsTable: React.FC = () => {
     completedCancelTask,
     failedCancelTask
   );
-  /*** Analysis */
-
-  const { tagItems } = useFetchTagsWithTagItems();
-
-  const [applicationDependenciesToManage, setApplicationDependenciesToManage] =
-    React.useState<Application | null>(null);
-  const isDependenciesModalOpen = applicationDependenciesToManage !== null;
 
-  const [assessmentToEdit, setAssessmentToEdit] =
-    React.useState<Assessment | null>(null);
-
-  const [reviewToEdit, setReviewToEdit] = React.useState<number | null>(null);
-
-  const [applicationsToDelete, setApplicationsToDelete] = React.useState<
-    Application[]
-  >([]);
-
-  const [assessmentToDiscard, setAssessmentToDiscard] =
-    React.useState<Application | null>(null);
-
-  const [reviewToDiscard, setReviewToDiscard] =
-    React.useState<Application | null>(null);
+  const cancelAnalysis = (application: DecoratedApplication) => {
+    const task = application.tasks.currentAnalyzer;
+    if (task?.id) cancelTask(task.id);
+  };
 
-  const [endOfAppImportPeriod, setEndOfAppImportPeriod] = useState<dayjs.Dayjs>(
-    dayjs()
-  );
+  const isTaskCancellable = (application: DecoratedApplication) => {
+    const task = application.tasks.currentAnalyzer;
+    return task?.state && !["Succeeded", "Failed"].includes(task.state);
+  };
 
+  // TODO: Review the refetchInterval calculation for the application list
   const {
-    data: applications,
+    data: baseApplications,
     isFetching: isFetchingApplications,
     error: applicationsFetchError,
   } = useFetchApplications(() =>
     hasActiveTasks || dayjs().isBefore(endOfAppImportPeriod) ? 5000 : false
   );
 
+  const {
+    applications,
+    applicationNames,
+    referencedArchetypeRefs,
+    referencedBusinessServiceRefs,
+  } = useDecoratedApplications(baseApplications, tasks);
```

Comment:
This is where all the decorated application data come in.  It's really the core of the work.